Widgets/WidgetViews.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  WidgetViews.swift
+//  DuckDuckGo
+//
+//  Copyright © 2020 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import WidgetKit
+import SwiftUI
+
+struct FavoriteView: View {
+
+    var favorite: Favorite?
+    var isPreview: Bool
+
+    var body: some View {
+
+        ZStack {
+            RoundedRectangle(cornerRadius: 10)
+                .fill(Color.widgetFavoritesBackground)
+                .isVisible(isPreview)
+
+            if let favorite = favorite, !isPreview {
+
+                RoundedRectangle(cornerRadius: 10)
+                    .fill(favorite.needsColorBackground ? Color.forDomain(favorite.domain) : Color.widgetFavoritesBackground)
+                    .shadow(color: Color.black.opacity(0.08), radius: 8, x: 0, y: 2)
+
+                Link(destination: favorite.url) {
+
+                    if let image = favorite.favicon {
+
+                        Image(uiImage: image)
+                            .scaleDown(image.size.width > 60)
+                            .cornerRadius(10)
+
+                    } else if favorite.isDuckDuckGo {
+
+                        Image("WidgetDaxLogo")
+                            .resizable()
+                            .frame(width: 45, height: 45, alignment: .center)
+
+                    } else {
+
+                        Text(favorite.domain.first?.uppercased() ?? "")
+                            .foregroundColor(Color.widgetFavoriteLetter)
+                            .font(.system(size: 42))
+
+                    }
+
+                }
+
+            }
+
+        }
+        .frame(width: 60, height: 60, alignment: .center)
+
+    }
+
+}
+
+struct LargeSearchFieldView: View {
+
+    var body: some View {
+        Link(destination: DeepLinks.newSearch) {
+            ZStack {
+
+                RoundedRectangle(cornerRadius: 21)
```

Comment:
This view could use `Capsule` instead of `RoundedRectangle` if the corner radius should be perfectly rounded on all corners, though I wasn't certain if `21` has been chosen specifically – ignore if so. 😄